validator/accounts/accounts_enable_disable_test.go

```diff
@@ -0,0 +1,144 @@
+
+	// We attempt to import accounts.
+	require.NoError(t, ImportAccountsCli(cliCtx))
+
+	// We attempt to disable the accounts specified.
+	require.NoError(t, DisableAccountsCli(cliCtx))
+
+	keymanager, err := w.InitializeKeymanager(cliCtx.Context, &iface.InitializeKeymanagerConfig{SkipMnemonicConfirm: false})
+	require.NoError(t, err)
+	remainingAccounts, err := keymanager.FetchValidatingPublicKeys(cliCtx.Context)
+	require.NoError(t, err)
+	require.Equal(t, len(remainingAccounts), 1)
+	remainingPublicKey, err := hex.DecodeString(k3.Pubkey)
+	require.NoError(t, err)
+	assert.DeepEqual(t, remainingAccounts[0], bytesutil.ToBytes48(remainingPublicKey))
+}
+
+func TestEnableAccounts_Noninteractive(t *testing.T) {
+	walletDir, _, passwordFilePath := setupWalletAndPasswordsDir(t)
+	randPath, err := rand.Int(rand.Reader, big.NewInt(1000000))
+	require.NoError(t, err, "Could not generate random file path")
+	// Write a directory where we will import keys from.
+	keysDir := filepath.Join(testutil.TempDir(), fmt.Sprintf("/%d", randPath), "keysDir")
+	require.NoError(t, os.MkdirAll(keysDir, os.ModePerm))
+
+	// Create 3 keystore files in the keys directory we can then
+	// import from in our wallet.
+	k1, _ := createKeystore(t, keysDir)
+	time.Sleep(time.Second)
+	k2, _ := createKeystore(t, keysDir)
+	time.Sleep(time.Second)
+	k3, _ := createKeystore(t, keysDir)
+	generatedPubKeys := []string{k1.Pubkey, k2.Pubkey, k3.Pubkey}
+	// Disable all keys.
+	disablePublicKeys := strings.Join(generatedPubKeys, ",")
+	// Only enable keys 0 and 1.
+	enablePublicKeys := strings.Join(generatedPubKeys[0:2], ",")
+	// We initialize a wallet with a imported keymanager.
+	cliCtx := setupWalletCtx(t, &testWalletConfig{
+		walletDir:           walletDir,
+		keymanagerKind:      keymanager.Imported,
+		walletPasswordFile:  passwordFilePath,
+		accountPasswordFile: passwordFilePath,
+		keysDir:             keysDir,
+		disablePublicKeys:   disablePublicKeys,
+		// Flags required for EnableAccounts to work.
+		enablePublicKeys: enablePublicKeys,
+	})
+	w, err := CreateWalletWithKeymanager(cliCtx.Context, &CreateWalletConfig{
+		WalletCfg: &wallet.Config{
+			WalletDir:      walletDir,
+			KeymanagerKind: keymanager.Imported,
+			WalletPassword: password,
+		},
+	})
+	require.NoError(t, err)
+
+	// We attempt to import accounts.
+	require.NoError(t, ImportAccountsCli(cliCtx))
+
+	// We attempt to disable the accounts specified.
+	require.NoError(t, DisableAccountsCli(cliCtx))
+
+	km, err := w.InitializeKeymanager(cliCtx.Context, &iface.InitializeKeymanagerConfig{SkipMnemonicConfirm: false})
+	require.NoError(t, err)
+	remainingAccounts, err := km.FetchValidatingPublicKeys(cliCtx.Context)
+	require.NoError(t, err)
+	require.Equal(t, len(remainingAccounts), 0)
+
+	// We attempt to enable the accounts specified.
+	require.NoError(t, EnableAccountsCli(cliCtx))
+
+	km, err = w.InitializeKeymanager(cliCtx.Context, &iface.InitializeKeymanagerConfig{SkipMnemonicConfirm: false})
```

Comment:
Is it normal that I have to call `InitializeKeyManager` again? I thought there was a go-routine that will check for any changes to the wallet/keymanager configs and will load that dynamically?

Reply:
Yes unfortunately :( because there is some important logic we have to do in NewKeymanager(), perhaps that code can be improved in a future PR